test/html.test.js

```diff
@@ -1073,6 +1086,147 @@ QUnit.test('translates metadata events into WebVTT cues', function() {
   QUnit.equal(cues[2].endTime, mediaSource.duration, 'sourceended is fired');
 });
 
+QUnit.test('removes existing metadata and caption tracks that exist on the player', function() {
+  let mediaSource = new videojs.MediaSource();
+  let sourceBuffer = mediaSource.addSourceBuffer('video/mp2t');
+
+  mediaSource.duration = Infinity;
+  mediaSource.nativeMediaSource_.duration = 60;
+
+  let addedTracks = [{
+    kind: 'metadata',
+    label: 'Timed Metadata'
+  }, {
+    kind: 'captions',
+    label: 'cc1'
+  }];
+  let removedTracks = [];
+  let metadata = [{
+    cueTime: 2,
+    frames: [{
+      url: 'This is a url tag'
+    }, {
+      value: 'This is a text tag'
+    }]
+  }, {
+    cueTime: 12,
+    frames: [{
+      data: 'This is a priv tag'
+    }]
+  }];
+
+  metadata.dispatchType = 0x10;
+  mediaSource.player_ = {
+    addRemoteTextTrack(options) {
+      let trackEl = {
+        track: {
+          kind: options.kind,
+          label: options.label,
+          addCue(cue) {}
+        }
+      };
+
+      addedTracks.push(trackEl.track);
+      return trackEl;
+    },
+    remoteTextTracks() {
+      return addedTracks;
+    },
+    removeRemoteTextTrack(track) {
+      removedTracks.push(track);
+      addedTracks.splice(addedTracks.indexOf(track), 1);
+    }
+  };
+
+  sourceBuffer.transmuxer_.onmessage(createDataMessage('video', new Uint8Array(1), {
+    metadata,
+    captions: [{
+      startTime: 1,
+      endTime: 3,
+      text: 'This is an in-band caption'
+    }]
+  }));
+  sourceBuffer.transmuxer_.onmessage(doneMessage);
+
+  QUnit.equal(removedTracks.length, 2, 'removed two text tracks');
+  QUnit.equal(removedTracks.filter(t => ['captions', 'metadata'].indexOf(t.kind) === -1).length,
+              0,
+              'removed only the expected two remote TextTracks');
+
+  QUnit.equal(addedTracks.length, 2, 'created two text tracks');
+  QUnit.equal(addedTracks.filter(t => ['captions', 'metadata'].indexOf(t.kind) === -1).length,
+              0,
+              'created only the expected two remote TextTracks');
```

Comment:
Maybe it's overboard, since the remove should be fine, but it might be worth checking to verify that the newly created tracks are the new ones.